src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -268,6 +287,10 @@ public boolean isSatisfiedBy(ByteBuffer columnValue)
 
         if (lower != null)
         {
+            // Post-filtering for LIKE operators must be consistent with implementation of LIKE operators in non-SAI Allow Filtering path
+            if(operator.isLikeVariant())
+                return termMatches(value.raw, lower.value.raw);
```

Comment:
I think we still may need to account for things like case-insensitive search eventually, so we should be able to just fall through to `validateStringValue()`, which would just call `termMatches()` anyway?